src/peft/peft_model.py

```diff
@@ -648,6 +650,8 @@ def _setup_prompt_encoder(self, adapter_name: str):
             prompt_encoder = PromptEncoder(config)
         elif config.peft_type == PeftType.PREFIX_TUNING:
             prompt_encoder = PrefixEncoder(config)
+        elif config.peft_type == PeftType.CPT:
+            prompt_encoder = CPTEmbedding(config, self.word_embeddings)
```

Comment:
Right now, this would just work even if the model is not a causal LM. Only later would users get an error. Let's add a check here. I think that `model.config.is_decoder` should work, though I'm not 100% sure. Let's raise an error if the model type is not correct and let's also add a test for that error.

Reply:
Could you please clarify where I should verify if the model type is correct? model.config.is_decoder is a property of the base model, right?

Reply:
Yes, the check would be `self.base_model.config.is_decoder`. You can check it at the start of the `_setup_prompt_encoder` method.

Reply:
Done